src/openrct2/ride/RideAudio.cpp

```diff
@@ -85,15 +87,8 @@ namespace OpenRCT2::RideAudio
             Pan = src.Pan;
             Frequency = src.Frequency;
 
-            if (Channel != nullptr)
-            {
-                Channel->Stop();
-            }
-            Channel = src.Channel;
-            src.Channel = nullptr;
-
-            Source = src.Source;
-            src.Source = nullptr;
+            swap(Channel, src.Channel);
+            swap(Source, src.Source);
```

Comment:
For the std library, we always use the full namespace rather than "using std".

Reply:
That's not it. By adding `using std::swap` which is not the same as `using namespace [X];` I let Argument Dependent Lookup figure out what `swap` function to pick - if types of `Channel` or `Source` end up getting their own specializations of `swap` (right now they don't, but this may change) then this syntax will automatically make use of them; else it falls back to `std::swap`.